xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/doc/DeletedDocumentRevisionProvider.java

```diff
@@ -58,14 +64,29 @@ public XWikiDocument getRevision(DocumentReference reference, String revision) t
         XWikiDeletedDocument deletedDocument = xcontext.getWiki().getDeletedDocument(Long.valueOf(revision), xcontext);
 
         // Only local the document if it matches the asked document reference
-        if (deletedDocument != null
-            && (reference == null || deletedDocument.getDocumentReference().equals(reference))) {
+        if (deletedDocument != null && areReferencesCompatible(reference, deletedDocument.getDocumentReference())) {
             return deletedDocument.restoreDocument(xcontext);
         }
 
         return null;
     }
 
+    private boolean areReferencesCompatible(DocumentReference requestedDocRef, DocumentReference deletedDocRef)
+    {
+        boolean result;
+        if (requestedDocRef == null) {
+            result = true;
```

Comment:
@tmortagne I kept this behaviour, but it feels a bit weird that we allow it. Our API in DocumentRevisionProvider is used with null references sometimes?

Reply:
Ok